mmdet/utils/util_distribution.py

```diff
@@ -0,0 +1,56 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import torch
+from mmcv.parallel import MMDataParallel, MMDistributedDataParallel
+
+USE_DEVICE = ''
+
+dp_factory = {
+        'cuda' : MMDataParallel
+    }
+
+ddp_factory = {
+        'cuda' : MMDistributedDataParallel
+    }
+
+def build_dp(model, device = 'cuda', device_ids = 0):
+    # build DataParallel module by device type.
+    # if device is cuda, return a MMDataParallel model;
+    # if device is mlu, return a MLUDataParallel model.
+    assert device in ['cuda', 'mlu'], "Only available for cuda or mlu devices."
+    if device == 'cuda':
+        model = model.cuda()
+    elif device == 'mlu':
+        from mmcv.device.mlu import MLUDataParallel
+        dp_factory['mlu'] = MLUDataParallel
+        model = model.mlu()
+
+    return dp_factory[device](model, device_ids=device_ids)
+
+
+def build_ddp(model, device = 'cuda', device_ids=0,
+        broadcast_buffers=False, find_unused_parameters=False):
+    # build DistributedDataParallel module by device type.
+    # if device is cuda, return a MMDistributedDataParallel model;
+    # if device is mlu, return a MLUDistributedDataParallel model.
+    assert device in ['cuda', 'mlu'], "Only available for cuda or mlu devices."
+    if device == 'cuda':
+        model = model.cuda()
+    elif device == 'mlu':
+        from mmcv.device.mlu import  MLUDistributedDataParallel
+        ddp_factory['mlu'] = MLUDistributedDataParallel
+        model = model.mlu()
+
+    return ddp_factory[device](model, device_ids=device_ids,
+                               broadcast_buffers=broadcast_buffers,
+                               find_unused_parameters=find_unused_parameters)
+
+def select_device():
+    global USE_DEVICE
+    if USE_DEVICE != '' :
+        return USE_DEVICE
+    is_device_available = {'cuda': torch.cuda.is_available(),
+                       'mlu': hasattr(torch, 'is_mlu_available') and torch.is_mlu_available()}
```

Comment:
1. suggest using a wrapper like below
```
def is_mlu_available():
    return hasattr(torch, 'is_mlu_available') and torch.is_mlu_available()
```
and then use `is_mlu_available()` here.
2.  seems that the global variable USE_DEVICE is not used in other places, can we unify the corresponding usages by using `select_device`?
3. select_device might not be an appropriate name, we can use `get_device()`

Reply:
ok， done